### PR TITLE
Invert comm map

### PR DESCRIPTION
**Pull request by lowrie**
* Improve performance of c4's invert_comm_map.
* Related to jayenne's #784

* [Pre-Merge Code Review](https://github.com/losalamos/Draco/wiki/Style-Guide)
  * [x] Travis CI checks pass
  * [x] Code coverage does not decrease
  * [x] [Valgrind test passes](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [ ] [Toss2 checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [ ] [Trinitite checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] Code reviewed/approved, sufficient DbC checks, testing, documentation

### CODE REVIEW

**Comment by KineticTheory** [comment]
@lowrie I don't fully understand the codecov-io metrics (especially when you update a PR after the first code coverage is reported).  However, in this case, you removed ~30 lines of code that were 100% covered.  These LOC helped bring the overall code average up.  So, removing them does indeed reduce the overall coverage percentage -- but there is nothing wrong with your code.

**Comment by KineticTheory** [comment]
@along Can you complete a code review on the MPI_Win features and testing?

**Comment by KineticTheory** [comment]
Cannot run tests on HPC because the machines are down.

**Comment by KineticTheory** [comment]
@lowrie @along Is this ready to be merged? Are there additional updates that you want to implement first?

File: src/c4/test/tstInvert_Comm_Map.cc [hunk 1]
```diff
@@ -73,47 +76,57 @@ void test4(rtt_c4::ParallelUnitTest &ut) {
...
+  if (static_cast<int>(from_nodes.size()) != nodes)
+    FAILMSG("Incorrect from_nodes size.");
+
   for (int i = 0; i < nodes; ++i) {
-    if (to_nodes.front() != from_nodes.front())
+    if (to_nodes[0] != from_nodes[0])
       FAILMSG("Incorrect data in map.");
-
-    to_nodes.pop_front();
-    from_nodes.pop_front();
```
Comment (KineticTheory) [inline src/c4/test/tstInvert_Comm_Map.cc]: @lowrie Alex thinks you need to leave these two `pop_front()` commands.

Comment (lowrie) [inline src/c4/test/tstInvert_Comm_Map.cc]: Oops, actually better I'll change the [0] to [i] on the earlier lines.

**Pull request merged**